File: Models/Post.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tyrell.Models
{
    public class Post
    {
        [Key]
        [ScaffoldColumn(false)]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long PostId {get;set;}
        public string Title {get;set;}
        public string Body {get;set;}
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime CreatedAt {get;set;}
        public bool IsVisible { get; set; }
        
        public long AuthorId { get; set; }
        public Author Author { get; set; }
        
//        public ICollection<string> Tags { get; set; }
    }
}

File: Client/src/app/Models/post.ts
export class Post {

  constructor(Title,AuthorId,Body,IsVisible) {
    this.title = Title;
    this.authorId = AuthorId;
    this.body = Body;
    this.isVisible = IsVisible;
  }

  postId: number;
  title: string;
  body: string;
  createdAt: Date;
  isVisible: boolean;

  authorId: number;
  author: Author;
}
